docs/changelog/v6.md

```diff
@@ -0,0 +1,214 @@
+
+Here's how you would execute a request-style and then a tell-style query in Halogen 5:
+
+```purs
+handleAction = do
+  a <- H.query _a unit (H.request Child.SomeRequestQuery)
+  _ <- H.query _a unit (H.tell Child.SomeTellQuery)
+```
+
+In Halogen 6, you no longer use the `query` function. Instead, you use `request` and `tell` directly. You also don't have to throw away the result of `tell`, as it can already be safely discarded:
+
+```purs
+handleAction = do
+  a <- H.request _a unit Child.SomeRequestQuery
+  H.tell _a unit Child.SomeTellQuery
+```
+
+The old `tell` and `request` functions still exist in Halogen 6, but they've been renamed to `mkTell` and `mkRequest` and are only used when querying the root of your application. For example, this code in Halogen 5:
+
+```purs
+io <- runUI component unit body
+
+state <- io.query $ H.request SomeRequestQuery
+_ <- io.query $ H.tell SomeTellQuery
+```
+
+becomes this code in Halogen 6:
+
+```purs
+io <- runUI component unit body
+
+state <- io.query $ H.mkRequest SomeRequestQuery
+_ <- io.query $ H.mkTell SomeTellQuery
+```
+
+Added in [#621](https://github.com/purescript-halogen/purescript-halogen/pull/621).
+
+## Event Sources
+
+Event sources have been implemented with `Event` from the `event` library instead of the Halogen event sources implementation on top of `coroutines`. This simplifies the library internals and connects Halogen with a more widely-used FRP library.
+
+Notable changes include:
+
+- The entire `Halogen.Query.EventSource` module has been removed and replaced with `Halogen.Query.Event` which provides only an `eventListenerEventSource` function. The new function is a drop-in replacement for the old one, so all you need to do is update your import.
+- `affEventSource` and `effectEventSource` functions can be trivially replaced with code using the `event` library directly, so they have been removed. Examples of how to rewrite these functions are below.
+- The other helper functions and types from the `Halogen.Query.EventSource` module are no longer required.
+- The `subscribe` function and `Subscribe` constructor no longer take an `EventSource m action` to subscribe to. They take an `Event action` instead.
+- The `HalogenIO` type returned by running your root-level component now contains a `messages :: Event output` instead of a `subscribe :: Coroutine.Consumer output m Unit -> m Unit`.
+
+If you were previously using `effectEventSource`, then you would change this Halogen 5 code:
+
+```purs
+import Halogen as H
+import Halogen.Query.EventSource as ES
+
+do
+  void $ H.subscribe $ ES.effectEventSource \emitter -> do
+    ES.emit emitter MyAction
+    pure mempty
+```
+
+with this Halogen 6 code:
+
+```purs
+import FRP.Event as Event
+import Halogen as H
+
+do
+  { event, push } <- H.liftEffect Event.create
+  void $ H.subscribe event
+  H.liftEffect $ push MyAction
+```
```

Comment:
Should there be two examples here? One that never unsubscribes and a second that does unsubscribe at some point?

Also, should this be rendered as a `diff` to better highlight what did change between the two examples?

Reply:
The entire body would be highlighted in the diff, and one of the imports, leaving only the `Halogen` import and the `do` lines untouched. For that reason I think it's better to have separate examples.